src/main/java/com/exasol/adapter/document/documentfetcher/files/randomaccessinputstream/RandomAccessInputStreamCache.java

```diff
@@ -0,0 +1,111 @@
+package com.exasol.adapter.document.documentfetcher.files.randomaccessinputstream;
+
+import java.io.IOException;
+
+/**
+ * This class is a wrapper for {@link RandomAccessInputStream}s that caches some data to avoid many small requests.
+ */
+public class RandomAccessInputStreamCache extends RandomAccessInputStream {
+    private final RandomAccessInputStream source;
+    private final byte[] cache;
+    private long cacheStartPos = 0;
+    private long usedCacheSize = 0;
+    private long position = 0;
```

Comment:
basically, the stream could be longer than the integer size. In that cases, it's important that they are longs. When accessing the cache it's however always integers, since java only supports integer parameters for array access.